frontend/src/components/Common/FirestoreDataProvider.tsx

```diff
@@ -36,6 +37,35 @@ export const useHasAdminPermission = (): boolean => {
   return self?.role === 'lead' || adminEmails.includes(userEmail);
 };
 
+export const useTeams = (): readonly Team[] => {
+  const allMembers = useMembers();
+  const teamsMap = new Map<string, Team>();
+
+  function getTeam(name: string) {
+    let team = teamsMap.get(name);
+    if (team != null) return team;
+    team = { uuid: name, name, leaders: [], members: [], formerMembers: [] };
```

Comment:
do we still need the `uuid` field for the `Team` type?

Reply:
We probably don't need it anymore, but I plan to migrate one thing at a time.